### PR TITLE
feat: add workflow entity

### PR DESCRIPTION
**Pull request by ronaldronson**
## Summary

Adding a new entity called Workflows under the alpha feature.

## Description

Adding typings and methods:
- create
- update
- query
- delete
- complete

No documentation update for alpha features.

## Checklist (check all before merging)

- [x] Both unit and integration tests are passing
- [x] There are no breaking changes
- [ ] Changes are reflected in the documentation

When adding a new method:

- [x] The new method is exported through the default and plain CMA client
- [x] All new public types are exported from `./lib/export-types.ts`
- [x] Added a unit test for the new method
- [x] Added an integration test for the new method
- [ ] The new method is added to the documentation

### CODE REVIEW

File: lib/adapters/REST/endpoints/workflow.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+) => {
+  const data = copy(rawData)
+  return raw.post<WorkflowProps>(http, getBaseUrl(params), data, {
+    headers,
+  })
+}
+
+export const update: RestEndpoint<'Workflow', 'update'> = (
+  http: AxiosInstance,
+  params: GetWorkflowParams,
```
Comment (Chaoste) [inline lib/adapters/REST/endpoints/workflow.ts:48]: Even though it's the same, that makes more sense :)

```suggestion
  params: UpdateWorkflowParams,
```

File: lib/entities/workflow.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+} from '../common-types'
+import { wrapCollection } from '../common-utils'
+import enhanceWithMethods from '../enhance-with-methods'
+
+export type WorkflowSysProps = Pick<
+  BasicMetaSysProps,
+  'id' | 'version' | 'createdAt' | 'createdBy' | 'updatedAt' | 'updatedBy'
+> & {
+  type: 'Workflow'
+  space: SysLink
```
Comment (Chaoste) [inline lib/entities/workflow.ts:21]: I'm wondering when do we use `SysLink` and when do we use an explicit type with `Link<T>`? I see that we do this in other places as well. Is there some pattern to it?

**Review by Chaoste — APPROVED** [review]
LGTM!
When bumping the CMA version in UI, also remember to pass this entity to the app sdk. [Here's how](https://github.com/contentful/experience-packages/pull/1417/files#diff-fb63c16adab012c7dc56b1b18581dd55e6ce72ddc5075b06edac76f2a257de81) we did it for the WorkflowDefinition.

File: lib/entities/workflow.ts (hunk 2 above)
Comment (ronaldronson) [reply lib/entities/workflow.ts:21]: Sys link doesn't enforce the type, as the Link does, in cases where it's less obvious by the key (like space) I think it makes more sense to be more restrictive and allow only a specific type.

**Pull request merged**